state/machine.go

```diff
@@ -2247,28 +2247,10 @@ func (m *Machine) SetUpgradeSeriesStatus(unitName string, status model.UnitSerie
 		if err := m.isStillAlive(); err != nil {
 			return nil, errors.Trace(err)
 		}
-		coll, closer := m.st.db().GetCollection(machineUpgradeSeriesLocksC)
-		defer closer()
-		var lock upgradeSeriesLockDoc
-		err := coll.FindId(m.Id()).One(&lock)
+		docIndex, err := m.getUnitIndex(unitName, status)
```

Comment:
@hmlanigan what do you think of routines like this which hide the extraction of the lock. Taking the lock as a parameter would alllow us to chain these things together at which point a little builder pattern would start to emerge, but I don't see that just yet since nothing yet calls for more than one of these little util methods. However, I was trying to hold off since the shape of the lock type may be forcing some things that could end up changing entirely. Don't want to jump the gun here.

Reply:
i like the little helper routines.